### PR TITLE
Only register validation metadata when PHPCR is enabled, adjust file structure

### LINKED ISSUE #277: Only register validation metadata when PHPCR is enabled

### PR DESCRIPTION
**Pull request by wouterj**
/fixes #277, #274

### CODE REVIEW

**Comment by wouterj** [comment]
Something went wrong here. b2a1627 was meant to be the only commit in this PR. But as there seems to be no issue created by 1b5dd70 , I propose to keep it in here.

Ready to merge imo. I don't agree with the style issue, currently looking for a way to disable this fixer.

File: .travis.yml [hunk 1]
```diff
@@ -10,20 +10,21 @@ sudo: false
...
 matrix:
   include:
-    - php: 7.0
-      env: DEPS=dev SYMFONY_VERSION=3.1.*
+    - php: 7.1
+      env: DEPS=dev SYMFONY_VERSION=3.3.*
     - php: 5.5
-      env: COMPOSER_FLAGS="--prefer-lowest" SYMFONY_DEPRECATIONS_HELPER=weak
-    - php: 5.6
-      env: DEPS=dev COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=3.0.*
```
Comment (wouterj) [inline .travis.yml]: Symfony 3.0 has reached end of maintaince, let's stop testing against it.

@symfony-cmf/core as we test against Symfony 2.8 and Symfony 3.1+, we can be very confident that we support 3.0. Shall we keep allowing it in composer.json, or shall we set it to `2.8.*|^3.1`?

**Comment by dbu** [comment]
unless we find reasons to want to drop support, we could keep testing it for a bit. does not cost much. allowing without testing seems risky

File: .travis.yml (hunk 1 above)
Comment (dbu) [inline .travis.yml]: i would prefer to keep building with 3.0 for now. if we find reasons to want to drop 3.0 support we can still do that. allowing it without testing seems wrong. if you don't agree, lets take this change out of this PR so we can merge it.

Comment (dbu) [inline .travis.yml]: #279

**Pull request merged**